modules/misunderstood/src/views/full/SideList.tsx

```diff
@@ -42,6 +43,22 @@ const SideList = ({
         </div>
       )}
 
+      {selectedStatus === FLAGGED_MESSAGE_STATUS.applied && events && events.length > 0 && (
+        <div className={style.applyAllButton}>
+          <Button onClick={deleteAllStatus(FLAGGED_MESSAGE_STATUS.applied)} intent={Intent.WARNING} icon="delete" fill>
+            {lang.tr('module.misunderstood.deleteAllDone')}
+          </Button>
+        </div>
+      )}
+
+      {selectedStatus === FLAGGED_MESSAGE_STATUS.deleted && events && events.length > 0 && (
+        <div className={style.applyAllButton}>
+          <Button onClick={deleteAllStatus(FLAGGED_MESSAGE_STATUS.deleted)} intent={Intent.WARNING} icon="delete" fill>
+            {lang.tr('module.misunderstood.deleteAllIgnored')}
+          </Button>
+        </div>
+      )}
+
```

Comment:
We are calculating 4 times `events && events.length > 0` and duplicating the code, can you refactor this part ?